### ISSUE TITLE
https://api.spotify.com/v1/me/player/next error

### ISSUE THREAD

**Issue by imminna** [spotify/web-api#859]
Issue found on May 2rd 2018.

#### Endpoint(s):
* `POST https://api.spotify.com/v1/me/player/next
--

#### Actual behaviour:
411 Server Error (consistently)
However I can't find the information about this error

**Comment by onemanbucket** [comment]
Hi!

Try setting `Content-Length: 0` as a header

**Comment by imminna** [comment]
Thanks, It's solved.
I did'nt set Content-Length: 0 as a header while using this API a month ago , however it works.
It seems this API make the revision.
could you please list the revision note while Spotify Web-API make the change ? THANKS!!

**Comment by michaelherger** [comment]
I do indeed see "411 Length Required". Please fix this issue or the documentation. Thank!

Thanks @onemanbucket - I'll give that a try. I'm wondering whether they expect length to be the seconds played on the previous song or similar. It's something I've seen for other streaming services' APIs.

**Comment by onemanbucket** [comment]
@michaelherger the `Content-Length` header is part of the HTTP spec and its value should be the length of the request body (in this case zero bytes).

**Comment by BloodShura** [comment]
+1. I'm also getting it for `/me/player/seek`. What's more strange is, the content returned is a Google-like HTML error page.

`Content-Length` was never required to be set before, and this sudden change can be actually breaking many services. So, either this is a temporary error, or this should really be stated in the Web-API documentation, as @michaelherger said.

**Comment by michaelherger** [comment]
Setting it to 0 did work for me, too.

To be fair these calls are still clearly labeled beta, with a warning that they could change any time. See https://beta.developer.spotify.com/documentation/web-api/reference/player/

**Comment by onemanbucket** [comment]
This is caused by an infrastructure change in the webapi load balancer, it's now behind a google https load balancer, which will reject illegal requests according to these rules: https://cloud.google.com/compute/docs/load-balancing/http/#illegal_request_handling

@BloodShura your request looks correct, can you provide a reproducable example and i'll investigate

**Comment by michaelherger** [comment]
Thanks @onemanbucket!

This might explain why _I_ was failing: https://stackoverflow.com/questions/1233372/is-an-http-put-request-required-to-include-a-body

**Comment by BloodShura** [comment]
@onemanbucket I managed to fix it!

If anyone else is also setting a `Content-Length: 0` header and is still experiencing this issue...

Behind the scenes, I'm using the Java's HTTP API (`java.net.HttpURLConnection`). By default, using `connection.setRequestProperty("Content-Length", 0)` will be ignored (unless `connection.setFixedLengthStreamingMode()` is called), because the `Content-Length` header is set automatically when you write data onto the connection's output stream.

The problem is: since some of these endpoints (like `/me/player/seek`) do not require anything to be sent in the body of the request (only in the query parameters), I, of course, wasn't writing anything to the connection's output stream, which caused the Java's API to not set the `Content-Length` header.

Then what I did was, convert the request to a "with-body" request (which will vary on which higher-level API you're using afterall), but without writing anything to the connection's output stream. This way, the output stream is opened, nothing is written, then the Java's API correctly sets a `Content-Length: 0` header.
